frontend/src/components/MapView/Legends/WfpWmsDownloadButton.tsx

```diff
@@ -0,0 +1,43 @@
+import { Button } from '@material-ui/core';
+import moment from 'moment';
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { WMSLayerProps } from '../../../config/types';
+import { dateRangeSelector } from '../../../context/mapStateSlice/selectors';
+import { useSafeTranslation } from '../../../i18n';
+import { DEFAULT_DATE_FORMAT } from '../../../utils/name-utils';
+import { downloadGeotiff, Extent } from '../Layers/raster-utils';
+
+interface IProps {
+  layer: WMSLayerProps;
+  extent?: Extent;
+}
+
+function WfpWmsDownloadButton({ layer, extent }: IProps) {
+  const { t } = useSafeTranslation();
+  const { startDate: selectedDate } = useSelector(dateRangeSelector);
+  const dispatch = useDispatch();
+
+  const handleDownload = () => {
+    downloadGeotiff(
+      layer.serverLayerName,
+      extent,
+      moment(selectedDate).format(DEFAULT_DATE_FORMAT),
+      dispatch,
+    );
+  };
+
+  return (
+    <Button
+      variant="contained"
+      color="primary"
+      size="small"
+      onClick={() => handleDownload()}
+      fullWidth
+    >
+      {t('Download WFP GeoTiff')}
```

Comment:
no need to say "wfp" here

Reply:
@wadhwamatic you recommended having wfp in the button name, didn't you?

Reply:
@sifelix - no... We talked about differentiating downloads delivered through STAC from non-STAC / WCS approaches, which would include WFP's Data Cube layers. That should be a config process, not something a user would ever see or know